### ISSUE TITLE
Locality Weighting?

### ISSUE THREAD

**Issue by candalfigomoro** [lmcinnes/umap#267]
In the docs you say

> And our results in terms of adjusted Rand score and adjusted mutual information are in line with the current state of the art techniques using convolutional autoencoder techniques. That’s not bad for an approach that is simply viewing the data as arbitrary 784 dimensional vectors.

In your opinion, would it be possible to add a concept of "locality" to UMAP? So for an image, you could give more weight to features (pixels) that are "spatially" close to each other

**Comment by lmcinnes** [comment]
Yes, this is certainly possible. Ultimately it comes down to writing a suitable custom distance metric. If you really want to play with things like this you might consider checking out the ``sinkhorn_distance`` in the 0.4dev branch, which implements an approximation to an earth-mover distance with the ground distance being l1 or l2 distance in the 2D pixel space.
